### ISSUE TITLE
Incorrect object location in .d files after ccache 3.7.5

### ISSUE THREAD

**Issue by carrotIndustries** [ccache/ccache#499]
### How to reproduce ###

1. create source file `foo.c` (see below, actual content doesn't really matter)
2. create `build directory`
2. run `ccache g++ -MP -MMD -c foo.cpp -o build/foo.o`
3. look at `build/foo.d`

foo.c:
```
#include <iostream>

int main(void) {
	std::cout << "hello world" << std::endl;
}
```

### Actual behavior ###

`build/foo.d` contains

```
foo.o: foo.cpp
```
Note that the the directory is missing from the object file.

### Expected behavior ###

`build/foo.d`should be
```
build/foo.o: foo.cpp
```

### Environment ###

ccache from master bc897d216fccc2ed2aeb7f3123d7a763694fedfd, but this behaviour was introduced by 19c3729d30640fc2c78242cb46136e619fdbd802 as this doesn't pass `-MQ` to the compiler any more and the compiler thus doesn't know about the object file's directory when invoked with `-E`. As far as I can tell, we should pass -MQ to the compiler unconditionally.

**Comment by jrosdahl** [comment]
Thanks for the bug report. It will be fixed in ccache 3.7.7.

**Comment by carrotIndustries** [comment]
Thanks for the quick fix!
